front/src/applications/operationalStudies/components/NGE/types.ts

```diff
@@ -91,17 +91,66 @@ export type TrainrunSectionDto = {
   warnings: unknown[];
 };
 
+/**
+ * Represents a train category in NGE (Netzgrafik Editor).
+ * Mandatory attribute for any Trainrun, used for filtering and visual representation (line color, style) of trains.
+ * Some attributes are specific to NGE and not used in OSRD synchronization.
+ */
 export type TrainrunCategory = {
+  /**
+   * Unique identifier for the category.
+   * Special cases:
+   * - id 0 is reserved for NO_CATEGORY
```

Comment:
I'd prefer to avoid adding some OSRD-specific details in this file if possible: this just describes the NGE side of things. The OSRD specifics are in MacroEditor.